packages/trader/src/Stores/Modules/Trading/trade-store.ts

```diff
@@ -1229,27 +1230,35 @@ export default class TradeStore extends BaseStore {
         // when accumulator is selected, we need to change chart type to mountain and granularity to 0
         // and we need to restore previous chart type and granularity when accumulator is unselected
         const {
-            prev_chart_type,
-            prev_granularity,
             chart_type,
             granularity,
-            savePreviousChartMode,
+            saveChartType,
+            saved_chart_type,
+            saved_granularity,
+            saveGranularity,
+            setChartTypeAndGranularity,
             updateChartType,
             updateGranularity,
         } = this.root_store.contract_trade || {};
-        if (isAccumulatorContract(obj_new_values.contract_type) || isDigitTradeType(obj_new_values.contract_type)) {
-            savePreviousChartMode(chart_type, granularity);
-            updateGranularity(0);
-            updateChartType('line');
-        } else if (
-            (obj_new_values.contract_type || obj_new_values.symbol) &&
-            prev_chart_type &&
-            prev_granularity &&
-            (prev_chart_type !== chart_type || prev_granularity !== granularity)
-        ) {
-            updateGranularity(prev_granularity);
-            updateChartType(prev_chart_type);
-            savePreviousChartMode('', null);
+        const has_line_chart =
+            isAccumulatorContract(obj_new_values.contract_type) ||
+            isDigitTradeType(obj_new_values.contract_type) ||
+            isUpDownContract(obj_new_values.contract_type);
+        if (!has_line_chart && (saved_chart_type || !Number.isNaN(saved_granularity))) {
```

Comment:
`if (!has_line_chart) {`
You are checking `saved_chart_type || !Number.isNaN(saved_granularity))` below

Reply:
Nothing wrong with condition, it works correct. I just left absolutely optional suggestion to modify it, but, one more time, it's optional)
The only thing is the video, where chart type in the url is not the same as aplied on the chart

Reply:
I mean your suggestion doesn't work the same way as mine, that's why I'm confused

Reply:
@kate-deriv  Regarding your second comment, there was a case when upon saving the chart_type and granularity into localStorage it wasn't updating query params correctly for default ticks contracts. I fixed it in my latest commit 👍🏻